lib/shopify_app/controller_concerns/login_protection.rb

```diff
@@ -170,7 +170,10 @@ def fullpage_redirect_to(url)
     end
 
     def current_shopify_domain
-      shopify_domain = sanitized_shop_name || session[:shopify_domain]
+      shopify_domain = sanitized_shop_name ||
+        jwt_shopify_domain ||
```

Comment:
Should there be a test for this case?

Reply:
It's a protected method. Do we wanna have test for protected methods @tanema ?

Reply:
Ideally I would like something to prove that this works. We have tests for full-page redirect so maybe you can add an extra case for full-page redirect that uses this logic?